wire/msgtx.go

```diff
@@ -554,9 +657,84 @@ func (msg *MsgTx) Copy() *MsgTx {
 	return &newTx
 }
 
+// writeTxInScriptsToMsgTx allocates the memory for variable length fields in a
+// MsgTx TxIns as a contiguous chunk of memory, then fills in these fields for the
+// MsgTx by copying to a contiguous piece of memory and setting the pointer.
+func writeTxInScriptsToMsgTx(msg *MsgTx, totalScriptSize uint64) {
```

Comment:
Splitting these reduces the effectiveness since it will end up with two allocations instead of one single allocation.  I would create a single func like `writeTxScriptsToMsgTx` or even `makeScriptsContiguous(msg *MsgTx, totalScriptSize uint64)`